### DISCUSSION TITLE
Slinger on Android

### DISCUSSION THREAD

**Discussion by GerryDazoo** [GerryDazoo/Slinger#216] [General]
Here's a screen shot of slingbox_server.py running on my Pixel6 using Pydroid3.
Installed Pydroid3 and installed Flask module. Made a folder and copied slingbox_server.py and config.ini. Told Pydroid3 to run it. Bob's your uncle.
A really mobile solution in your pocket.

![image](https://user-images.githubusercontent.com/102599361/199520572-6d0f889b-897b-4dd6-b196-98cfd04401c8.png)

**Comment by surveyor2** [comment]
It's fine in terms of working on a local LAN but would not be a mobile solution unless it can find the slingboxes if you were to go outside.

I tried Process Alive from
 https://singularlabs.com/forums/topic/processalive-automatically-restart-crashed-programs/
To look after my sling server at home on windows while I was out and the remote server restart from the remote menu worked fine.

That isn't to say that the server crashed. It didn't. But just in case it does require a remote reset or someone accidentally presses the reset button on the remote while they're out it's useful freebie to work alongside the slingserver to keep it running.

James

**Reply by GerryDazoo** [reply]
No, problem you just give the server the external IP addresses of the slingboxes. Please read the Networking Guide for details. If your on linux setup a service on windows run the supplied batch file to restart a crashed server. No special code required.

**Reply by surveyor2** [reply]
Oh okay. It'll be interesting to see how it works out in the real world (4g/5g blindspots etc). The LAN based server was very solid though. I'm not brave enough just yet to try a remote server. I've got remotely accessible power sockets on all the slings just in case they misbehave :)

James

**Reply by sp222222** [reply]
James, what brand type do you use? Were they expensive?

**Reply by flipper2222222** [reply]
> James, what brand type do you use? Were they expensive?

I use D-Link
SKU: DSP-W118 
D-Link Mini Wi-Fi Smart Plug

**Reply by surveyor2** [reply]
> James, what brand type do you use? Were they expensive?

I use Meross. https://www.amazon.co.uk/gp/product/B07QX6CF46/ref=ppx_yo_dt_b_asin_title_o01_s00?ie=UTF8&psc=1
For no other reason than they were listed as Amazon choice, quite cheap and had a good reliability rating. They haven't disappointed so far. I can't say if they're any better or worse than others though 

James

**Comment by GerryDazoo** [comment]
Well, this might be interesting but until I can find out why pydroid3 is shutting down my process after a few minutes it's a non-starter. I haven't been able to find a way to preserve the logs after the shutdown to see what's happening because PD closes the window immediately.
